hbase_sql/hbase1.1.0/src/main/java/com/splicemachine/hbase/BackupEndpointObserver.java

```diff
@@ -85,31 +101,39 @@ public void prepareBackup(
-                }
-                // check again if the region is being split. If so, return an error
-                if (isSplitting) {
-                    if (LOG.isDebugEnabled()) {
-                        SpliceLogUtils.debug(LOG, "%s:%s is being split when trying to prepare for backup", tableName, regionName);
-                    }
-                    responseBuilder.setReadyForBackup(false);
-                    //delete the ZNode
-                    ZkUtils.recursiveDelete(path);
-                } else {
-                    //wait for all compaction and flush to complete
-                    if (LOG.isDebugEnabled()) {
-                        SpliceLogUtils.debug(LOG, "%s:%s waits for flush and compaction to complete", tableName, regionName);
+                if (!BackupUtils.regionIsBeingBackup(tableName, regionName, path)) {
+                    canceled = BackupUtils.backupCanceled();
+                    if (!canceled) {
+                        HBasePlatformUtils.flush(region);
+                        // Create a ZNode to indicate that the region is being copied
+                        ZkUtils.recursiveSafeCreate(path, HConfiguration.BACKUP_IN_PROGRESS, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
                     }
-                    region.waitForFlushesAndCompactions();
-                    if (LOG.isDebugEnabled()) {
-                        SpliceLogUtils.debug(LOG, "%s:%s is ready for backup", tableName, regionName);
+                }
+                if (!canceled) {
+                    // check again if the region is being split. If so, return an error
+                    if (isSplitting) {
+                        if (LOG.isDebugEnabled()) {
+                            SpliceLogUtils.debug(LOG, "%s:%s is being split when trying to prepare for backup", tableName, regionName);
+                        }
+                        responseBuilder.setReadyForBackup(false);
+                        //delete the ZNode
+                        ZkUtils.recursiveDelete(path);
+                    } else {
+                        //wait for all compaction and flush to complete
+                        if (LOG.isDebugEnabled()) {
+                            SpliceLogUtils.debug(LOG, "%s:%s waits for flush and compaction to complete", tableName, regionName);
+                        }
+                        region.waitForFlushesAndCompactions();
+                        if (LOG.isDebugEnabled()) {
+                            SpliceLogUtils.debug(LOG, "%s:%s is ready for backup", tableName, regionName);
+                        }
+                        responseBuilder.setReadyForBackup(true);
                     }
-                    responseBuilder.setReadyForBackup(true);
                 }
+                else
+                    responseBuilder.setReadyForBackup(false);
             }
```

Comment:
this part of the code look like duplicate from the one above, , is there a reason to not have it encapsulate in a same component like a class or a method ?